service/create/launch_configuration.go

```diff
@@ -115,19 +119,40 @@ func (s *Service) createLaunchConfiguration(input launchConfigurationInput) (boo
 }
 
 func (s *Service) deleteLaunchConfiguration(input launchConfigurationInput) error {
+	groupName := securityGroupName(input.cluster.Name, input.prefix)
+	sg := awsresources.SecurityGroup{
+		Description: groupName,
+		GroupName:   groupName,
+		AWSEntity:   awsresources.AWSEntity{Clients: input.clients},
+	}
+
+	sgID, err := sg.GetID()
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	workersLCName, err := launchConfigurationName(input.cluster, prefixWorker, sgID)
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
 	lc := awsresources.LaunchConfiguration{
-		Name: input.name,
+		Client: input.clients.AutoScaling,
+		Name:   workersLCName,
 	}
 
 	if err := lc.Delete(); err != nil {
 		return microerror.MaskAny(err)
 	}
-	s.logger.Log("debug", fmt.Sprintf("deleted launch configuration '%s'", input.name))
-
 	return nil
 }
 
-func launchConfigurationName(cluster awstpr.CustomObject, prefix string) (string, error) {
+// launchConfigurationName uses the cluster ID, a prefix and the security group
+// ID to produce a launch configuration name.  LC names are their unique
+// identifiers in AWS.  The reason we need the securityGroupID in the name is
+// that we can only reuse an LC if it has been created for the current SG.
+// Otherwise, the SG might not exist anymore.
```

Comment:
OK that answers my question on the sec group ID ;)

I guess for multiple ASGs we could add the instance type to the name. But that is a later change so we can address it then.